vaadin-list-box-flow-parent/vaadin-list-box-flow/src/main/java/com/vaadin/flow/component/listbox/MultiSelectListBox.java

```diff
@@ -66,8 +66,16 @@ public MultiSelectListBox() {
     private static <T> JsonArray modelToPresentation(
             MultiSelectListBox<T> listBox, Set<T> model) {
         JsonArray array = Json.createArray();
-        model.stream().map(listBox.getItems()::indexOf)
-                .forEach(index -> array.set(array.length(), index));
+        model.stream()
```

Comment:
I believe this change should be covered by a new unit test in `ListBoxUnitTest/MultiSelectListBoxTest` with different identifier cases.
And also for other changes where `listBox.getItemId()` is used as a replacement of `equals()`

Reply:
This should be followed up and fixed within another ticket, so as to not hold the component alpha1 release.